### PR TITLE
fixed (internal) not working sometimes

### PR DESCRIPTION
**Pull request by SimonSchick**
There was some weirdness going on with the patching on raml objects, we should rewrite this in TS entirely, but for now this makes sure we don't leak internal docs anymore.

### CODE REVIEW

File: build/docs.js [hunk 1]
```diff
@@ -143,16 +143,19 @@ function traverseResources (ramlObj, parentUrl, allUriParameters) {
...
     const newTypes = {};
-    ramlObj.types.forEach(type => _.assign(newTypes, type));
-    ramlObj.types = orderObject(newTypes);
+    types
+    .forEach(type => _.assign(newTypes, type));
 
-    return ramlObj;
+    return orderObject(_.omitBy(newTypes, type =>
+        (type.annotations && type.annotations.internal))
+    );
```
Comment (connor4312) [inline build/docs.js]:
```js
return orderObject(
  _(types)
  .reduce((out, type) => _.assign(out, type), {})
  .omitBy(type => type.annotations && type.annotations.internal)
  .value()
);
```

File: build/docs.js [hunk 2]
```diff
@@ -176,36 +179,32 @@ function addUniqueIdsToDocs (ramlObj) {
...
-    ramlObj = transverseTypes(ramlObj);
+    ramlObj.types = transverseTypes(ramlObj.types);
     return addUniqueIdsToDocs(ramlObj);
 }
 
 /**
  * Generates a custom raml2html config object, injecting some of our own logic
  * to add features raml2html doesn't provide by default.
+ * @param {Resource[]} resource
+ * @param {Function} callback
```
Comment (connor4312) [inline build/docs.js]: `predicate` or something would be more accurate

File: build/docs.js [hunk 3]
```diff
@@ -176,36 +179,32 @@ function addUniqueIdsToDocs (ramlObj) {
...
+ * @param {absUrl} string
  * @return {Object}
  */
 function traverseRAMLResourceTree (resources, callback, absURL) {
     absURL = absURL || '';
```
Comment (connor4312) [inline build/docs.js]: default args

File: build/docs.js (hunk 1 above)
Comment (SimonSchick) [inline build/docs.js]: That doesn't work :c

Comment (connor4312) [inline build/docs.js]: oh :c

**Pull request merged**